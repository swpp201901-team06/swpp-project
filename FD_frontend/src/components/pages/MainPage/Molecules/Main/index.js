import React, { PropTypes } from 'react'
import styled from 'styled-components'
import { font, palette } from 'styled-theme'
import MainButton from '../../Atoms/MainButton'

const Wrapper = styled.div`
  font-family: ${font('primary')};
  color: ${palette('grayscale', 0)};
`

export const Main = ({ statefunction, onClickSignin, onClickGuest }) => {
  const onSignin = () => {
    console.log("si_mol");
    onClickSignin();
  };
  const onGuest = () => {
    console.log("gu_mol");
    onClickGuest();
  };
  return (
    <div>
      <MainButton type="signin" onClick={onSignin}>로그인</MainButton>
      <MainButton type="guest" onClick={onGuest}>게스트</MainButton>
    </div>
  );
}

Main.propTypes = {
  reverse: PropTypes.bool,
  children: PropTypes.node,
}
