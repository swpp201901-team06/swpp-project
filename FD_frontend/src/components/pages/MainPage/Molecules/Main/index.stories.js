import React from 'react'
import { storiesOf } from '@kadira/storybook'
import { Main } from 'components'

storiesOf('Main', module)
  .add('default', () => (
    <Main>Hello</Main>
  ))
  .add('reverse', () => (
    <Main reverse>Hello</Main>
  ))
