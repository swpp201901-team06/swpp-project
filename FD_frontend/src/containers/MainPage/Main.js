import { connect } from 'react-redux'
import { Main } from '../../components/pages/MainPage/Molecules/Main'
import { goToSignin, goToGuest } from '../../store/Mainpage/actions'
import { baseHistory } from '../../index'

const mapStateToProps = (state) =>{
    return {
        statefunction : state
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        onClickSignin: () => {
            baseHistory.push('/signin')
            dispatch(goToSignin())
        },
        onClickGuest: () => {
            baseHistory.push('/guest')
            dispatch(goToGuest())
        }
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Main)
