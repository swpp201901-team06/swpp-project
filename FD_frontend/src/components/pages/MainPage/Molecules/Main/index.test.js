import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Main from './index'

describe('Main molecule test', () => {
  const onClickSignIn = jest.fn()
  const onClickGuest = jest.fn()
  const wrap = (props = {}) => shallow(<Main {...props} />)

  beforeEach(() => {
    onClickSignIn.mockReset()
    onClickGuest.mockReset()
  })

  it('matches snapshot', () => {
    const renderedValue = renderer.create(<Main />).toJSON()
    expect(renderedValue).toMatchSnapshot()
  })

  it('renders Main', () => {
    const wrapper = wrap()
    expect(wrapper.length).toEqual(1)
  })

  it('renders two MainButton\'s', () => {
    const wrapper = wrap()
    expect(wrapper.find('MainButton')).toHaveLength(2)
    expect(wrapper.find('MainButton').at(0).childAt(0).text()).toMatch('Login')
    expect(wrapper.find('MainButton').at(1).childAt(0).text()).toMatch('Guest')
  })

  it('calls onClickSignIn on clicking Sign In button', () => {
    const props = {
      onClickSignIn,
      onClickGuest,
    }
    const wrapper = wrap(props)
    wrapper.find('MainButton').at(0).simulate('click')
    expect(onClickSignIn).toHaveBeenCalled()
  })

  it('calls onClickGuest on clicking Guest button', () => {
    const props = {
      onClickSignIn,
      onClickGuest,
    }
    const wrapper = wrap(props)
    wrapper.find('MainButton').at(1).simulate('click')
    expect(onClickGuest).toHaveBeenCalled()
  })
})
