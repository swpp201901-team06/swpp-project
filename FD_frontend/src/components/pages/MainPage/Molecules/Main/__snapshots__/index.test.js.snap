// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main molecule test matches snapshot 1`] = `
<div>
  <button
    className="bnZcBb"
    height={100}
    onClick={[Function]}
    style={
      Object {
        "margin": 10,
      }
    }
    type="signin"
  >
    Login
  </button>
  <button
    className="bnZcBb"
    height={100}
    onClick={[Function]}
    style={
      Object {
        "margin": 10,
      }
    }
    type="guest"
  >
    Guest
  </button>
</div>
`;
